### ISSUE TITLE
db schema: BenchmarkResult: review data/times nullable

### ISSUE THREAD

**Issue by jgehrcke** [conbench/conbench#812]
From here: https://github.com/conbench/conbench/pull/811/files#r1127737995

The Column is marked as Nullable. But I wonder: why would we want to store null/None instead of an emtpy list? What's the semantic difference between both situations? Maybe we only need the empty list. In that case we remove the Nullable, and the type is correct.

**Comment by austin3dickey** [comment]
I agree that it's better to have a non-nullable column with a typed "empty" default. But one possible reason for why it looks the way it does today is that these columns could have been added to the schema later, once data already existed in production databases.

Since the Arrow database has 30 million rows in the `benchmark_result` table, it is difficult to add a non-nullable column to that table. In the migration, you must add the column as nullable, UPDATE all the existing rows to have a default value, and then change the column to non-nullable. That UPDATE may take 24+ hours, according to @ElenaHenderson . From Slack: 

<img width="387" alt="image" src="https://user-images.githubusercontent.com/16600275/224123807-d6574eea-c9c7-48d6-a15b-b3e4c252c0b3.png">

That said, I can't imagine that there are many rows in the Arrow DB where the value of these columns is actually None. I wonder if a migration using UPDATE WHERE would execute in a relatively short timeframe.
